sql/item_sum.cc

```diff
@@ -3553,6 +3553,57 @@ String *Item_sum_udf_str::val_str(String *str)
 #endif /* HAVE_DLOPEN */
 
 
+Item_sum_str::Item_sum_str(THD *thd)
+  : Item_sum(thd)
+{
+}
+
+Item_sum_str::Item_sum_str(THD *thd, Item_sum_str *item)
+  : Item_sum(thd, item)
+{
+}
+
+Item_sum_str::Item_sum_str(THD *thd, Item *item_par)
+  : Item_sum(thd, item_par)
+{
+}
+
+
+bool
+Item_sum_str::fix_fields(THD *thd, Item **ref)
+{
+  DBUG_ASSERT(fixed() == 0);
+
+  if (init_sum_func_check(thd))
+    return TRUE;
+
+  set_maybe_null();
+
+  /*
+    Fix fields for select list and ORDER clause
+  */
+
+  for (uint i=0 ; i < arg_count ; i++)
+  {
+    if (args[i]->fix_fields_if_needed_for_scalar(thd, &args[i]))
+      return TRUE;
+    /* We should ignore FIELD's in arguments to sum functions */
+    with_flags|= (args[i]->with_flags & ~item_with_t::FIELD);
+  }
+
+  bool ffi_result= fix_fields_impl(thd, ref);
+
+  if (ffi_result)
+      return ffi_result;
+
```

Comment:
The variable ffi_result does not seem to be needed.

```
if (fix_fields_impl(thd, ref))
  return true;
```
Wouldn't this work instead?